neo4j/_sync/driver.py

```diff
@@ -392,6 +392,89 @@ def supports_multi_db(self):
             return session._connection.supports_multiple_databases
 
+    def query(self, query, parameters=None, **kwargs):
+        """
+        :param query: cypher query
+        :type query: str, neo4j.Query
+        :param parameters: dictionary of parameters
+        :type parameters: dict
+        :param kwargs: additional keyword parameters
+        :returns: a new :class:`neo4j.QueryResult` object
+        :rtype: QueryResult
+        """
+        session_kwargs = {}
+        if "database" in kwargs:
+            session_kwargs["database"] = kwargs.pop("database")
+
+        with self.session(**session_kwargs) as session:
+            return session.query(query, parameters, **kwargs)
+
+    def execute(self, transaction_function, *args, **kwargs):
+        """Execute a unit of work in a managed transaction.
+
+        .. note::
+            This does not necessarily imply access control, see the session
+            configuration option :ref:`default-access-mode-ref`.
+
+        This transaction will automatically be committed unless an exception is thrown during query execution or by the user code.
+        Note, that this function perform retries and that the supplied `transaction_function` might get invoked more than once.
+
+        Managed transactions should not generally be explicitly committed
+        (via ``tx.commit()``).
+
+        Example::
+
+            def do_cypher_tx(tx, cypher):
+                records, _ = tx.query(cypher)
+                return records
+
+            values = driver.execute(do_cypher_tx, "RETURN 1 AS x")
+
+        Example::
+
+            def do_cypher_tx(tx):
+                records, _ = tx.query("RETURN 1 AS x")
+                return records
+
+            values = driver.execute(do_cypher_tx,
+                database="neo4j",
+                cluster_member_access=neo4j.api.CLUSTER_READERS_ACCESS)
+
+        Example::
+
+            def get_two_tx(tx):
+                result = tx.run("UNWIND [1,2,3,4] AS x RETURN x")
+                values = []
+                for record in result:
+                    if len(values) >= 2:
+                        break
+                    values.append(record.values())
+                # or shorter: values = [record.values()
+                #                       for record in result.fetch(2)]
+
+                # discard the remaining records if there are any
+                summary = result.consume()
+                # use the summary for logging etc.
+                return values
+
+
+            values = driver.execute(get_two_tx)
+
+        :param transaction_function: a function that takes a transaction as an
+            argument and does work with the transaction.
+            `transaction_function(tx, *args, **kwargs)` where `tx` is a
+            :class:`.Transaction`.
+        :param args: arguments for the `transaction_function`
+        :param kwargs: key word arguments for the `transaction_function`
```

Comment:
The docs use [reStructuredText](https://www.writethedocs.org/guide/writing/reStructuredText/) not markdown (I wish they would :upside_down_face:)
```suggestion
            argument and does work with the transaction.
            ``transaction_function(tx, *args, **kwargs)`` where ``tx`` is a
            :class:`.Transaction`.
        :param args: arguments for the ``transaction_function``
        :param kwargs: key word arguments for the ``transaction_function``
```